src/cargo/ops/cargo_test.rs

```diff
@@ -141,9 +141,20 @@ fn run_doc_tests(
     let config = options.compile_opts.config;
 
     // We don't build/run doc tests if `target` does not equal `host`.
-    if compilation.host != compilation.target {
-        return Ok((Test::Doc, errors));
-    }
+    let runtool = match (
+        compilation.target != compilation.host,
+        compilation.target_runner(),
+    ) {
+        (true, Some(x)) => Some(x),
+        (true, None) => {
+            config.shell().status(
+                "Doc-tests",
+                "Skipping doctests, no runner defined for target",
+            )?;
```

Comment:
This is not correct, it should just run without runner (like is done for `cargo run`)

Reply:
Good point

Reply:
And also it should probably use the runner regardless of whether target == host since that is what cargo run does